glib/src/shared.rs

```diff
@@ -527,7 +527,7 @@ impl<T, MM: SharedMemoryManager<Target = T>> Hash for Shared<T, MM> {
 
 impl<'a, T: 'static, MM> ToGlibPtr<'a, *mut T> for Shared<T, MM>
 where
-    MM: SharedMemoryManager<Target = T> + 'static,
+    MM: SharedMemoryManager<Target = T> + 'a,
```

Comment:
This should probably go into a separate commit (in the same PR). Is the same change needed for `src/boxed.rs` too, for example?

Reply:
right, we need that for https://github.com/gtk-rs/gtk4-rs/issues/1280

Reply:
This change is no longer part of this PR, since I now implement the reference counting manually instead of using `SharedMemoryManager`/`Shared<P, MM>` due to variance (https://github.com/gtk-rs/gtk-rs-core/pull/1225#discussion_r1396678845)